compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -451,7 +451,15 @@ public PathParameterSymbol createPathParamSymbol(BVarSymbol symbol, PathSegment.
         if (symbol == null) {
             return null;
         }
-        return new BallerinaPathParameterSymbol(kind, symbol, this.context);
+        return new BallerinaPathParameterSymbol(kind, symbol, this.context, false);
+    }
+
+    public PathParameterSymbol createPathParamSymbol(BTypeSymbol tsymbol, PathSegment.Kind kind) {
+        if (tsymbol == null) {
+            return null;
```

Comment:
Add a test to cover this

Reply:
@pcnfernando sure. Will do.

Reply:
Was this done?

Reply:
Is there any case where this can be null? IMO, we should always get a `tSymbol`.

Reply:
Removed the null check in https://github.com/ballerina-platform/ballerina-lang/pull/38146/commits/f453590e22a044b18b221b0842bf2d518279ea1a since the `tSymbol` is always set.